src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,79 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
+            AcquireTokenForClientParameters clientParameters, 
+            AcquireTokenCommonParameters commonParameters, 
+            RequestContext requestContext)
+        {
+            if (!clientParameters.UseMtlsPop)
+                return;
+
+            ValidateClaimsOrAssertionsUse();
+            ValidateAuthorityType();
+            ValidateCertificatePresence();
+            ValidateRegionPresence(requestContext);
+
+            commonParameters.MtlsCertificate = _confidentialClientApplication.Certificate;
+            commonParameters.AuthenticationOperation = new MtlsPopAuthenticationOperation(_confidentialClientApplication.Certificate);
+            requestContext.ServiceBundle.Config.IsInstanceDiscoveryEnabled = false;
+            ServiceBundle.Config.ClientCredential = null;
+            requestContext.UseMtlsPop = true;
```

Comment:
Why do we need this if the `AuthenticationOperation` is set to `MtlsPopAuthenticationOperation` ?